addon/components/inputs/select.js

```diff
@@ -116,81 +134,95 @@ export default AbstractInput.extend({
-    const modelDef = this.get('bunsenModel')
-    const options = _.get(cellConfig, 'renderer.options')
+  /**
+   * Determine if query has references to properties that have yet to be set.
+   * If the query contains no references then there is nothing to wait on.
+   * @param {Object} formValue - form value
+   * @returns {Boolean} whether or not query is waiting on missing references
+   */
+  iswaitingOnReferences (formValue) {
+    const {bunsenId, bunsenModel, cellConfig} = this.getProperties(
+      'bunsenId', 'bunsenModel', 'cellConfig'
+    )
 
-    if (options) {
-      return _.assign({}, options, modelDef)
-    }
+    const {query} = getMergedOptions(bunsenModel, cellConfig)
 
-    return modelDef
+    return (
+      typeOf(query) === 'object' &&
+      !hasValidQueryValues(formValue, query, bunsenId)
+    )
   },
 
   /* eslint-disable complexity */
   formValueChanged (newValue) {
-    const modelDef = this._getModelDef()
-    const oldValue = this.get('formValue')
+    const {
+      bunsenModel,
+      cellConfig,
+      formValue: oldValue
+    } = this.getProperties('bunsenModel', 'cellConfig', 'formValue')
+
+    const options = getMergedOptions(bunsenModel, cellConfig)
+
     this.set('formValue', newValue)
 
-    if (!modelDef) {
+    if (!options) {
       return
     }
 
-    const isQueryDisabled = this.isQueryDisabled(newValue)
+    const isWaitingOnReferences = this.iswaitingOnReferences(newValue)
 
-    if (this.get('queryDisabled') !== isQueryDisabled) {
-      this.set('queryDisabled', isQueryDisabled)
+    if (this.get('waitingOnReferences') !== isWaitingOnReferences) {
+      this.set('waitingOnReferences', isWaitingOnReferences)
     }
 
-    if (!isQueryDisabled && this.hasQueryChanged(oldValue, newValue, modelDef.query) || this.needsInitialOptions()) {
-      // setting required variables once above condition is true
-      const store = this.get('store')
-      const bunsenId = this.get('bunsenId')
-      const listData = this.get('listData')
-      // prevent multiple api calls when multiple formValueChanged is fired before options has a chance to be set
-      this.set('optionsInitialized', true)
-      listUtils.getOptions(newValue, modelDef, listData, bunsenId, store)
-        .then((opts) => {
-          this.set('options', opts)
-        })
-        .catch((err) => {
-          this.onError(bunsenId, [{
-            path: bunsenId,
-            message: getErrorMessage(err)
-          }])
-        })
+    // If we are still waiting on missing query references then there is nothing
+    // to do at this point
+    if (isWaitingOnReferences) {
+      return
     }
-  },
 
-  hasQueryParams (query) {
-    if (!query || _.isEmpty(query)) {
-      return false
+    // If the query has changed or we have yet to initialize the items lets go
+    // get our items
+    if (
+      this.hasQueryChanged(oldValue, newValue, options.query) ||
+      this.needsInitialItems()
```

Comment:
I fear that evaluating `this.needsInitialItems()` after `isWaitingOnReferences` will prevent the initial fetch for selects that don't have references but have query defined.